internal/settings/cache.go

```diff
@@ -99,3 +106,19 @@ func (s *Settings) LocationsCache() (map[string]string, string) {
 	}
 	return locations, defaultLocation
 }
+
+const TOKEN_VALID_CACHE_KEY_PREFIX = "token_valid."
+
+func (s *Settings) SetTokenValidCache(token string, exp int64) {
+	key := TOKEN_VALID_CACHE_KEY_PREFIX + strings.ReplaceAll(token, ".", "_")
+	setCacheWithExp(key, exp, true)
+}
+
+func (s *Settings) TokenValidCache(token string) bool {
+	key := TOKEN_VALID_CACHE_KEY_PREFIX + strings.ReplaceAll(token, ".", "_")
+	ok, err := getCache[bool](key)
+	if err != nil {
+		return false
+	}
+	return ok
```

Comment:
Ultra nit: This is just `return err != nil && ok`

Reply:
Will do! But you almost got me there 😬